### PR TITLE
Extract reduce_sum contribution

### LINKED ISSUE #181: reduce_sum bug

### PR DESCRIPTION
**Pull request by morgangiraud**
Closes #181

### CODE REVIEW

File: tensorflow_encrypted/protocol/pond.py [hunk 1]
```diff
@@ -421,6 +421,9 @@ def sum(self, x, axis, keepdims):
 
         return func(self, x, axis, keepdims)
 
+    def reduce_sum(self, x, axis=0, keepdims=False):
```
Comment (mortendahl) [inline tensorflow_encrypted/protocol/pond.py]: should we make default values matching TF or will that complicate matters? (eg https://www.tensorflow.org/api_docs/python/tf/reduce_sum)

File: tensorflow_encrypted/protocol/pond.py [hunk 2]
```diff
@@ -818,9 +821,12 @@ def add(self, other):
     def __add__(self, other):
         return self.prot.add(self, other)
 
-    def sum(self, axis, keepdims=False):
+    def sum(self, axis=0, keepdims=False):
```
Comment (mortendahl) [inline tensorflow_encrypted/protocol/pond.py]: if TF only has `reduce_sum` then maybe we should simply follow that to keep it simple?

Comment (morgangiraud) [reply tensorflow_encrypted/protocol/pond.py]: depends if you want to be friendly with numpy users?

File: tensorflow_encrypted/protocol/pond.py (hunk 1 above)
Comment (morgangiraud) [reply tensorflow_encrypted/protocol/pond.py]: I couldn't due on the complicated implementation of `crt_sum` under the hood. But if we replace everything with `tf.reduce_sum` under the hood then yes.
@jvmancuso ?

File: tensorflow_encrypted/protocol/pond.py (hunk 2 above)
Comment (mortendahl) [reply tensorflow_encrypted/protocol/pond.py]: wdyt?

Comment (mortendahl) [reply tensorflow_encrypted/protocol/pond.py]: could we swap so that `reduce_sum` is called by `sum` at least? (okay this is vanity, I know..)

Comment (morgangiraud) [reply tensorflow_encrypted/protocol/pond.py]: I don't mind having aliases, but it needs to be mentioned in the doc so people don't think it's a different behavior under the hood.

Comment (mortendahl) [reply tensorflow_encrypted/protocol/pond.py]: that was my concern as well. my vote is on sticking with TF (since we're promoting that anyway) but okay with catering a bit for numpy users as well if you guys think it's worth it

File: tensorflow_encrypted/protocol/pond.py [hunk 3]
```diff
@@ -1538,24 +1544,24 @@ def _sum_core(prot: Pond,
...
 
 def _sum_masked(prot: Pond,
                 x: PondMaskedTensor,
-                axis: int,
-                keepdims: Optional[bool]) -> PondPrivateTensor:
+                axis: Optional[int] = None,
+                keepdims: Optional[bool] = None) -> PondPrivateTensor:
     # y_on_0, y_on_1 = _sum_core(prot, x.unmasked, axis, keepdims)
     # return PondPrivateTensor(prot, y_on_0, y_on_1)
     raise NotImplementedError
```
Comment (mortendahl) [inline tensorflow_encrypted/protocol/pond.py]: for masked values, just forward to the unmasked values: `prot.sum(x.unmasked, .., ..)`

File: tests/test_reduce_sum.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import unittest
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_encrypted as tfe
+
+
+class Testconcat(unittest.TestCase):
```
Comment (jvmancuso) [inline tests/test_reduce_sum.py]: From #183, we should rename this

Comment (morgangiraud) [reply tests/test_reduce_sum.py]: Good catch, I hate when people don't change it ... Now I'm hating myself 🥇
